sdks/java/core/src/test/java/org/apache/beam/sdk/io/CountingSourceTest.java

```diff
@@ -128,6 +135,28 @@ public void testBoundedSourceSplits() throws Exception {
   }
 
   @Test
+  @Category({RunnableOnService.class, UsesAttemptedMetrics.class})
```

Comment:
It might make more sense for these tests to be in some Metrics-specific location, rather than associated with CountingSource. But, I don't have a strong opinion.